_posts/2020-10-27-mutiny-back-pressure.adoc

```diff
@@ -0,0 +1,314 @@
+image::/assets/images/posts/mutiny-back-pressure/back-pressure-drop.png[]
+
+You can drop the newest received items or oldest ones:
+
+[source, java, indent=0]
+----
+Multi.createFrom().ticks().every(Duration.ofMillis(10))
+    .onOverflow().drop(x -> System.out.println("Dropping item " + x))
+    .emitOn(Infrastructure.getDefaultExecutor())
+    .onItem().transform(DropExample::canOnlyConsumeOneItemPerSecond)
+    .transform().byTakingFirstItems(10)
+    .subscribe().with(
+        item -> System.out.println("Got item: " + item),
+        failure -> System.out.println("Got failure: " + failure)
+);
+----
+
+[source, text, indent=0]
+----
+// ....
+Dropping item 997
+Dropping item 998
+Dropping item 999
+Dropping item 1000
+Dropping item 1001
+Dropping item 1002
+Dropping item 1003
+Got item: 9
+----
+
+Dropping items provides a sustainable solution to our problem, but we are losing events!
+As we can see in the previous output, we may drop the high majority of items.
+In many cases, this is not acceptable.
+
+So, we need another solution, a solution where the overall pace is adjusted to satisfy the pipeline's slowest element. 
+We need flow control.
+
+### What the heck is Back-Pressure?
+
+You may have seen this term many times, and often associated with Reactive. 
+In mechanics, back-pressure is a way to control the flow of fluid through pipes, leading to a pressure drop.
+That control can use reducers or bends. 
+While very interesting, if you are a plumber, it's not clear how it can help us here? 
+
+We can see our streams as a flow of fluid, and the set of stages (operator or subscriber) forms a pipe. 
+We want to make the fluid flows as frictionless as possible without swirls and waves. 
+
+An interesting characteristic of fluid mechanics is how a downstream reduction of the throughput affects the upstream.  
+Essentially, that's what we need: a way for the downstream operators and subscribers to reduce the throughput, not only locally but also upstream. 
+
+Don't be mistaken; back-pressure is not something new in the IT world and it not limited to Reactive. 
+One of the most brilliant usages of back-pressure is in TCP.
+A reader, receiving data, can block the writer, on the other side of the wire, if it does not read the sent data.
+That way, the reader is never overwhelmed. 
+But, the consequence need to be understood: blocking the writer may not be without side-effects.
+
+There are other back-pressure implementations. 
+AMQP uses a credit-based flow control, where you can only send if you have enough credit.
+Eclipse Vert.x back-pressure is based on pause/resume, where a consumer can pause the upstream until it catches up and resume it once back on track. 
+
+### Introducing Reactive Streams
+
+Let's now focus on another back-pressure protocol: https://www.reactive-streams.org/[Reactive Streams].
+It defines an asynchronous and back-pressure protocol suiting perfectly to our fast producer/slow consumer problem.
```

Comment:
Maybe reword "suiting perfectly to our fast", seems a little off.

Maybe "suited" is enough?